nima/webclient/tracing/src/main/java/io/helidon/nima/webclient/tracing/WebClientTracing.java

```diff
@@ -0,0 +1,169 @@
+
+    private WebClientTracing() {
+        tracerFunction = ctx -> ctx.get(Tracer.class)
+                .orElseGet(Tracer::global);
+    }
+
+    private WebClientTracing(Tracer tracer) {
+        this.tracerFunction = ctx -> tracer;
+    }
+
+    /**
+     * Creates new instance of client tracing service.
+     *
+     * @return client tracing service
+     */
+    public static WebClientService create() {
+        return new WebClientTracing();
+    }
+
+    /**
+     * Creates a new client tracing service for a specific tracer.
+     *
+     * @param tracer to create new spans
+     * @return client tracing service
+     */
+    public static WebClientService create(Tracer tracer) {
+        return new WebClientTracing(tracer);
+    }
+
+    @Override
+    public WebClientServiceResponse handle(Chain chain, WebClientServiceRequest request) {
+        String method = request.method().text();
+
+        Tracer tracer = tracerFunction.apply(request.context());
+
+        Span.Builder spanBuilder = tracer.spanBuilder(composeName(method, request));
+
+        request.context().get(SpanContext.class).ifPresent(spanBuilder::parent);
+
+        spanBuilder.kind(Span.Kind.CLIENT);
+        spanBuilder.tag(Tag.COMPONENT.create("helidon-webclient"));
+        spanBuilder.tag(Tag.HTTP_METHOD.create(method));
+        spanBuilder.tag(Tag.HTTP_URL.create(request.uri().toString()));
+        Span span = spanBuilder.start();
+
+        request.context().register(span.context());
+
+        tracer.inject(span.context(),
+                      HeaderProvider.empty(),
+                      new ClientHeaderConsumer(request.headers()));
+
+        try {
+            WebClientServiceResponse response = chain.proceed(request);
+            Http.Status status = response.status();
+            span.tag(Tag.HTTP_STATUS.create(status.code()));
+
+            Http.Status.Family family = status.family();
+            if (family == Http.Status.Family.CLIENT_ERROR || family == Http.Status.Family.SERVER_ERROR) {
+                span.addEvent("error", Map.of("message",
+                                              "Response HTTP status: " + status,
+                                              "error.kind",
+                                              (family == Http.Status.Family.CLIENT_ERROR)
+                                                      ? "ClientError"
+                                                      : "ServerError"));
+            }
+
+            span.end();
+
+            return response;
+        } catch (Throwable e) {
+            span.end(e);
+            throw e;
+        }
+    }
+
+    private String composeName(String method, WebClientServiceRequest request) {
+        UriHelper uri = request.uri();
+        return method
+                + "-"
+                + uri.scheme() + "://"
+                + uri.host() + ":"
+                + uri.port()
+                + uri.path();
+    }
```

Comment:
We have same construct already in the `UriHelper.toString()`
```suggestion
    private String composeName(String method, WebClientServiceRequest request) {
        return method + "-" + request.uri();
    }
```

Reply:
Using `toString()` for this kind of features is not good, as the contract of `toString()` is different (it should be informational, not specified). 
I would prefer to have an option to modify `toString()` without expecting other components to break.
Will fix, but differently